File: 326.power-of-three.py
#
# @lc app=leetcode id=326 lang=python3
#
# [326] Power of Three
#

# @lc code=start
class Solution:
    def isPowerOfThree(self, n: int) -> bool:
        return n in {1, 3, 9, 27, 81, 243, 729, 2187, 6561, 19683, 59049, 177147, 531441, 1594323, 4782969, 14348907, 43046721, 129140163, 387420489, 1162261467, 3486784401, 10460353203, 31381059609, 94143178827, 282429536481, 847288609443, 2541865828329, 7625597484987, 22876792454961, 68630377364883, 205891132094649, 617673396283947}
    
        
# @lc code=end

if __name__ == "__main__":
    res = []
    for i in range(32):
        res.append(3**i)
    print(res)

File: golang/326.power-of-three.go
/*
 * @lc app=leetcode.cn id=326 lang=golang
 *
 * [326] Power of Three
 */
package main

// @lc code=start
func isPowerOfThree(n int) bool {
	if n <= 0 {
		return false
	}
	max := 1
	for max*3 < 1<<32 {
		max *= 3
	}
	return max%n == 0
}

// @lc code=end
